packages/client/src/devices/InputMediaDeviceManager.ts

```diff
@@ -74,10 +74,10 @@ export abstract class InputMediaDeviceManager<
    * Starts stream.
    */
   async enable() {
+    this.state.prevStatus = this.state.status;
```

Comment:
I think, you can replace that with `this.state.prevStatus = this.state.optmisticStatus`. Then the check in lines 101-108 will not be required.

I'm not 100% sure though, and the PR LGTM anyway.